changelogs/unreleased/5699-yangyy93-small.md

```diff
@@ -0,0 +1 @@
+Gateway configuration program: Add the `overloadMaxHeapSize` configuration option to contourDeployment to allow adding [overloadManager](https://projectcontour.io/docs/main/config/overload-manager/) configuration when generating envoy's initial configuration file
```

Comment:
Just a minor nit: could you change Gateway configuration program to Gateway provisioner as that has been used in the past as a prefix for changelog entries.